### PR TITLE
fix(worker): emit session.idled when act tasks hit DLQ

### PR DESCRIPTION
**Pull request by chaliy**
## What
When a durable `act` task exhausts all retries (DLQ) after a `reason` activity emitted server-backed tool calls, `emit_dlq_error_event` would silently return early, skipping the user-facing error message, `session.idled` event, and the `session status → idle` update. The session stayed stuck in the active state and the UI never unblocked.

Fixes [EVE-306](https://linear.app/everruns/issue/EVE-306).

## Why
`emit_dlq_error_event` parsed `task.input` as `DurableTurnInput` only. `act` tasks store their input as `ActTaskInput` (top-level `org_id` + flattened `ActInput`), so the parse always failed for them. An in-code comment claimed a fallback existed; it didn't. The control-plane already surfaces act-task DLQs via `update_workflow_status(Completed, Some(error))`, but the user-facing events were lost.

## How
- Extract a `TaskSessionContext` helper + `extract_task_session_context(activity_type, input)` that parses the correct wrapper shape for each activity type (`process_input` / `reason` → `DurableTurnInput`, `act` → `ActTaskInput`).
- Switch `emit_dlq_error_event` to use the helper and consume the extracted `org_id` for `set_session_status` (identical semantics to before, now just reachable for act tasks too).
- Add regression tests: parsing succeeds for reason, parsing succeeds for act (EVE-306 guard), parsing returns `None` for unknown activity types.

Single-file change in `crates/worker/src/durable_worker.rs` (+131/-10). No behavior change for the non-DLQ code paths.

## Risk
- Low
- Failure path only. If the helper somehow misparses valid input, the previous behavior — silently returning — is what we still get. Regression tests cover the exact serialization shapes the worker enqueues today (`ActTaskInput` with flattened `org_id` + `ActInput`, and `DurableTurnInput`).

## Security
- Threat categories reviewed: **TM-TENANT**. The fix now reaches the existing `set_session_status(org_id, session_id, "idle")` call for act tasks. `org_id` is the same trusted value the worker used when it enqueued the act task (`ActTaskInput.org_id`), which already flows through gRPC org validation per TM-TENANT-007. No new data flow, no new trust boundary, no cross-org exposure.
- Findings and resolutions: None. No new attack surface.

## Checklist
- [x] Tests added or updated (3 new regression tests in `durable_worker::tests`)
- [x] Backward compatibility considered (internal code; no external API change)
- [x] Security review performed against relevant threat model categories
- [ ] All review comments addressed (code change or written reasoning)

## Validation
- `cargo test -p everruns-worker --lib` → 87 passed, 0 failed
- `cargo build -p everruns-worker` clean
- `./scripts/lib/pre-push.sh` → all checks pass (fmt, clippy, lockfile, migrations, author)
- Worker binary boots cleanly and enters expected gRPC retry loop when control-plane is absent
- Full DLQ path requires NATS+PG+failing-tool infra; covered by the three new unit tests that verify the serialization shapes produced by the current scheduling code

### CODE REVIEW

File: crates/worker/src/durable_worker.rs [hunk 1]
```diff
@@ -43,6 +43,54 @@ struct ActTaskInput {
...
 }
 
+// =============================================================================
+// Task Session Context Extraction
+// =============================================================================
+
+/// Session context extracted from a task input.
+///
+/// Used by failure paths (DLQ emission, cancellation) that need to surface
+/// user-facing events regardless of which phase (input/reason/act) failed.
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/worker/src/durable_worker.rs]: Updated the doc comment to match actual usage (`DLQ emission and other task-input error paths`) in commit e0da07a, matching your suggestion. Refactoring the cancellation path to share the helper would be a larger scope change — left as a follow-up since cancellation also needs to fall back to a workflow-derived session_id when parse fails, which doesn't fit the helper's contract.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: crates/worker/src/durable_worker.rs [hunk 2]
```diff
@@ -1058,14 +1106,14 @@ impl DurableWorker {
...
+        // Extract session context from the task input. `act` task input is
+        // wrapped in `ActTaskInput`, while `reason`/`process_input` use
+        // `DurableTurnInput`; both shapes must be handled or act-task DLQs
+        // skip emitting session.idled and leave the session stuck.
+        let Some(ctx) = extract_task_session_context(&task.activity_type, &task.input) else {
             warn!(
                 task_id = %task.id,
+                activity_type = %task.activity_type,
                 "Cannot emit DLQ error event: failed to parse task input"
             );
```
[the remark this answers is not in the document]
Comment (chaliy) [reply crates/worker/src/durable_worker.rs]: Good catch. In commit e0da07a I changed `extract_task_session_context` to return `Result<TaskSessionContext>` with `anyhow::Context` wrapping the underlying serde error per activity type, and updated `emit_dlq_error_event` to log the full chain via `format!("{err:#}")`. Added a regression test (`test_extract_task_session_context_act_malformed_includes_parse_error`) that asserts the wrapping context survives.

---
_Generated by [Claude Code](https://claude.ai/code)_
